dash/_patch.py

```diff
@@ -0,0 +1,117 @@
+def _operation(name, location, **kwargs):
+    return {"operation": name, "location": location, "params": dict(**kwargs)}
+
+
+_noop = object()
+
+
+class Patch:
+    """
+    Patch a callback output value
+
+    Act like a proxy of the output prop value on the frontend.
+
+    Supported prop types: Dictionaries and lists.
+    """
+
+    def __init__(self, location=None, parent=None):
+        if location is not None:
+            self._location = location
+        else:
+            # pylint: disable=consider-using-ternary
+            self._location = (parent and parent._location) or []
+        if parent is not None:
+            self._operations = parent._operations
+        else:
+            self._operations = []
+
+    def __getitem__(self, item):
+        return Patch(location=self._location + [item], parent=self)
```

Comment:
`__getitem__` and `__setitem__` can receive `slice` objects - I can see that being useful, for inserting or removing elements in the middle of an array, but it generates all sorts of edge cases that might make intuitive sense but not sure we really want to support. I'm thinking of stuff like `Patch()[1::2].val += 1` which I'd interpret as "go into all the odd items and increment their `val` properties by 1"

So maybe for now the safest thing to do is just raise an error if you get here with a `slice`?

Reply:
Also, do `__getitem__` and `__setitem__` work with negative indices?

Reply:
No negative index in js, so no.

Reply:
Can we try to make that work? Feels like it would be really useful for "edit the last thing I added" type of operations.